### ISSUE TITLE
Plugin suggestion

### ISSUE THREAD

**Issue by glappen** [jaap-karssenberg/zim-desktop-wiki#291]
Hello, I am am new zim wiki user and have been enjoying it a lot.  I just had a need for a feature which I did not see build in or in any plugins - it would be nice to be able to make a page that is encrypted, and that requires a password to view.  Each page could have it's own password, or the user could have one password to decrypt everything. But it would prevent someone access the computer and reading secure information in a page.  How hard would that be?  I have some python experience and would be interested in maybe programming this.

Thanks!

**Comment by Osndok** [comment]
I've thought about this a lot, actually, but it generally revolves around these few facts:
* whole-disk encryption & home-directory encryption at least partially ameliorate this as an individual's need (so what remains for discussion is really collaborative & backup)
* encrypted backup is probably a concern better left for another project (so what remains is collaboration)
* strictly speaking, per-page encryption would surely end up being too annoying for a collaborative use case, so you need to think about applying access to a **class** of pages to a **group** of users, or maybe even control access to a whole notebook, or even have a particular branch of a zim repo simply removed (such as by a git submodule) and control access that way (such as with gitolite)
* One might want to encrypt only **PART** of a page, such as to hide a password, have an official redaction, or the like.

**Comment by Osndok** [comment]
Attachments are also a concern.

**Comment by glappen** [comment]
Yes, being able to encrypt a part of a page would be useful, perhaps a more
generic solution.

**Comment by synchr** [comment]
I don't know if this is exactly what you are looking for, but there already exists a plugin to encrypt a selection of text: https://github.com/jaap-karssenberg/zim-wiki/wiki/Crypt-Selection-Plugin

I haven't tried it though.

**Comment by angerangel** [comment]
I use Zim Wiki and I'd pay for a plugin that automatically lock and encrypt a page and request a password to unlock it. I use it on Windows

**Comment by bflorat** [comment]
Same here, I can pay 100€ to get this feature (ideally, one page password, we try to open, a page, a dialog ask for a password, AES 256).

BTW, I never managed to make the Crypt Selection Plugin be detected by zim (Ubuntu package).

Thx for this great tool
